client/src/components/add-job-post-component/job-info-form/index.jsx

```diff
@@ -0,0 +1,252 @@
+        />
+
+        {/* Employer Name Field */}
+        <TextField
+          fullWidth
+          sx={{ m: 1, width: "96%" }}
+          id="employer"
+          label="Employer Name"
+          value={jobPostData.employer}
+          onChange={(e) => handleInputChange(e.target.value, "employer")}
+          helperText="*Required"
+          required
+          InputLabelProps={{ required: false }}
+        />
+
+        {/* Location Field */}
+        <TextField
+          fullWidth
+          sx={{ m: 1, width: "96%" }}
+          id="location"
+          label="Location"
+          value={jobPostData.location}
+          onChange={(e) => handleInputChange(e.target.value, "location")}
+          helperText="*Required"
+          required
+          InputLabelProps={{ required: false }}
+        />
+
+        {/* Minimum Compensation Field */}
+        <FormControl fullWidth sx={{ m: 1, width: "47%" }}>
+          <InputLabel id="minCompensationLabel">
+            Minimum Compensation
+          </InputLabel>
+          <OutlinedInput
+            id="minCompensationLabel"
+            type="number"
+            startAdornment={<InputAdornment position="start">$</InputAdornment>}
+            label="Minimum Compensation"
+            inputProps={{ min: 0 }}
+            value={jobPostData.rate_of_pay_min}
+            onChange={(e) => {
+              const { value } = e.target;
+              if (/^\d*\.?\d*$/.test(value))
+                handleInputChange(value, "rate_of_pay_min");
+            }}
+            required
+            InputLabelProps={{ required: false }}
+          />
+        </FormControl>
+
+        {/* Maximum Compensation Field */}
+        <FormControl fullWidth sx={{ m: 1, width: "47%" }}>
+          <InputLabel id="maxCompensationLabel">
+            Maximum Compensation
+          </InputLabel>
+          <OutlinedInput
+            id="maxCompensationLabel"
+            type="number"
+            inputProps={{ min: 0 }}
+            startAdornment={<InputAdornment position="start">$</InputAdornment>}
+            label="Maximum Compensation"
+            value={jobPostData.rate_of_pay_max}
+            onChange={(e) => {
+              const { value } = e.target;
+              if (/^\d*\.?\d*$/.test(value))
+                handleInputChange(value, "rate_of_pay_max");
+            }}
+            required
+            InputLabelProps={{ required: false }}
+          />
+        </FormControl>
+
+        {/* Compensation Rate Field */}
+        <FormControl fullWidth sx={{ m: 1, width: "47%" }}>
+          <InputLabel id="compensationRateLabel">Compensation rate</InputLabel>
+          <Select
+            sx={{ textAlign: "left" }}
+            labelId="compensationRateLabel"
+            id="compensationRateLabel"
+            value={jobPostData.rate_of_pay_frequency}
+            label="Compensation rate"
+            onChange={(e) =>
+              handleInputChange(e.target.value, "rate_of_pay_frequency")
```

Comment:
I think here the e.target.value might be "Base & Commission" if the option is selected and if the API for adding a job post is called it might result in an error due to ENUM types not being able to contain the ampersand (&), but I'm not entirely sure though.